Src/WindowsServer/WindowsServer.Shared/AppServicesHeartbeatTelemetryModule.cs

```diff
@@ -0,0 +1,150 @@
+        {
+            AppServicesHeartbeatTelemetryModule.Instance = this;
+        }
+
+        /// <summary>
+        /// Gets a value to provide internal access to the only instance of this class that *should* be available.
+        /// </summary>
+        internal static AppServicesHeartbeatTelemetryModule Instance
+        {
+            get => AppServicesHeartbeatTelemetryModule.instance;
+            private set => AppServicesHeartbeatTelemetryModule.instance = value;
+        }
+
+        /// <summary>
+        /// Initialize the default heartbeat provider for Azure App Services. This module
+        /// looks for specific environment variables and sets them into the heartbeat 
+        /// properties for Application Insights, if they exist.
+        /// </summary>
+        /// <param name="configuration">Unused parameter.</param>
+        public void Initialize(TelemetryConfiguration configuration)
+        {
+            // Core SDK creates 1 instance of a module but calls Initialize multiple times
+            if (!this.isInitialized)
+            {
+                lock (this.lockObject)
+                {
+                    if (!this.isInitialized)
+                    {
+                        this.AddAppServiceEnvironmentVariablesToHeartbeat(this.GetHeartbeatPropertyManager(), false);
+
+                        this.isInitialized = true;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Ensure we've cleaned up our static Instance.
+        /// </summary>
+        public void Dispose()
+        {
+            // ensure our Instance variable is not kept around.
+            AppServicesHeartbeatTelemetryModule.instance = null;
+        }
+
+        /// <summary>
+        /// Signal the AppServicesHeartbeatTelemetryModule to update the values of the 
+        /// Environment variables we use in our heartbeat payload.
+        /// </summary>
+        internal void UpdateHeartbeatWithAppServiceEnvVarValues()
+        {
+            if (this.isInitialized)
+            {
+                this.AddAppServiceEnvironmentVariablesToHeartbeat(this.GetHeartbeatPropertyManager(), isUpdateOperation: true);
+            }
+        }
+
+        private IHeartbeatPropertyManager GetHeartbeatPropertyManager()
+        {
+            IHeartbeatPropertyManager hbeatManager = null;
+            var telemetryModules = TelemetryModules.Instance;
+
+            try
+            {
+                foreach (var module in telemetryModules.Modules)
+                {
+                    if (module is IHeartbeatPropertyManager hman)
+                    {
+                        hbeatManager = hman;
+                    }
+                }
+            }
+            catch (Exception hearbeatManagerAccessException)
+            {
+                WindowsServerEventSource.Log.HeartbeatManagerAccessFailure(hearbeatManagerAccessException.Message);
+            }
+
+            return hbeatManager;
```

Comment:
You do not check for `null` here. It may be a good idea though

Reply:
Agreed. Done.